Project.toml

```diff
@@ -17,6 +17,8 @@ ReverseDiff = "37e2e3b7-166d-5795-8a7a-e32c996b4267"
 
 [compat]
 ADNLPModels = "0.3.1"
+ExpressionTreeForge = "0.1.0"
+julia = "^1.6.0"
 JuMP = "0.21.5"
 MathOptInterface = "0.9.22"
```

Comment:
and this?

Reply:
I don't really see how NLPModelsJuMP and OptimizationProblems are dependencies either.

Reply:
`NLPModelsJuMP` :
- alos used by `get_expr_tree(::MathOptNLPModel)`
- If we change `get_expr_tree(::ADNLPModel)` to `get_expr_tree(::AbstractNLPModel)`, we have to keep `get_expr_tree(::MathOptNLPModel)`, otherwise there will be errors during the execution.

`OptimizationProblems` is an extra dependency, I informed a compatibility because the tests use an old version of  `OptimizationProblems`.
I will modify the tests now and remove the compat in a new PR.

Reply:
Some packages can be dependencies for the tests, and that's ok. But I doubt OptimizationProblems is an actual dependency of the package.

Reply:
`get_expr_tree` must go look into the JuMP model. Do you *really* need MathOptNLPModel??? It's just a wrapper.

Reply:
> Some packages can be dependencies for the tests, and that's ok. But I doubt OptimizationProblems is an actual dependency of the package.

OptimizationProblems is not in [deps] only on [extra], the [compat] was to force the good version for the tests.

> `get_expr_tree` must go look into the JuMP model. Do you _really_ need MathOptNLPModel??? It's just a wrapper.
Right now, I use it to have the same `get_expr_tree`  method with different behaviors if the argument is an MathOptNLPModel or a ADNLPModel.

That way, for any `NLPModel::Union{ADNLPModel, MathOptNLPModel}` argument, I apply `get_expr_tree` and apply the partial-separability detection on `get_expr_tree` result.
I planned to extend in the future `SupportedNLPModel = Union{ADNLPModel, MathOptNLPModel}` with more `NLPModel`s.

Reply:
The #31 upgrade the version of OptimizationProblems in the deps.
I add a commit to remove OptimizaitonProblems from [compat].

Reply:
upgrade?

Reply:
> upgrade?

I used OptimizationProblems v.0.3.0 when it implemented the tests.
```julia
n = 10
nlp = MathOptNLPModel(OptimizationProblems.arwhead(n), name = "arwhead " * string(n))
```
I changed it to v0.4.3:
```julia
n = 10
nlp = arwhead(; n)
```

Reply:
Implemented in #31